### ISSUE TITLE
internal error: ASSERTION FAILED: file rts/sm/GCAux.c, line 44

### ISSUE THREAD

**Issue by tmcdonell** [AccelerateHS/accelerate#339]
Affects at least both LLVM backends.

**Comment by tmcdonell** [comment]
This may or may not be the same error: it did get much further than usual before failing, and [GHC#11108](https://ghc.haskell.org/trac/ghc/ticket/11108) implies that the above should be fixed in 8.0.1.

```
internal error: ASSERTION FAILED: file (null), line 1690

    (GHC version 8.0.1.20161117 for x86_64_apple_darwin)
    Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```

**Comment by tmcdonell** [comment]
@mchakravarty @robeverest As mentioned earlier today, [GHC#11108](https://ghc.haskell.org/trac/ghc/ticket/11108) seems to be the relevant bug in GHC-7.10. The seems to trigger the bug in fairly short order (after a few thousand operations only):

```
accelerate-nofib --llvm-cpu +RTS -N -RTS
```

(That uses _N_ threads per run but also runs _N_ tests concurrently. May seem like a lot but the oversubscription is worthwhile since the vast majority of time is spent in the reference code / equality tests, not in accelerate. The problem still arises without running tests concurrently.)

As that ticket mentions you can (it seems) work around it with `+RTS -G1` to disable GC generations, but productivity drops dramatically; for `nofib` (probably a worst-case since most of the reference implementations are via lists) to < 10%, so maybe this is just hiding the problem since so little actual work is being done.

**Comment by mchakravarty** [comment]
@tmcdonell Ok, thanks. I'll look at this once I have got the LLVM compilation/installation sorted.

**Comment by tmcdonell** [comment]
Seems to be fixed in ghc-8.0.2.
